src/estimagic/optimization/tranquilo/options.py

```diff
@@ -27,13 +27,18 @@ def get_default_sample_size(model_type, x):
 
 def get_default_model_fitter(model_type, sample_size, x):
     n_params = n_free_params(dim=len(x), model_type=model_type)
-    if model_type == "linear":
-        fitter = "ridge" if sample_size < n_params else "ols"
+    if model_type == "linear" or sample_size >= n_params:
+        fitter = "ols"
     else:
         fitter = "tranquilo"
     return fitter
 
 
+def get_default_residualize(model_type, sample_size, x):
+    n_params = n_free_params(dim=len(x), model_type=model_type)
+    return not (model_type == "linear" or sample_size >= n_params)
```

Comment:
Should we merge these two functions to express that `residualize` is by default used iff the tranquilo fitter is used?

Reply:
Or make `default_residualize` dependent on fitter